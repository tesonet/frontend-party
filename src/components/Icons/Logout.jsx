import React from 'react';

import { IconWrapper } from './atoms';

const Logout = props => (
  <IconWrapper {...props}>
    <svg
      xmlns='http://www.w3.org/2000/svg'
      width='16'
      height='16'
      viewBox='0 0 16 16'
      fill='currentColor'
    >
      <path
        id='Combined_shape_7742'
        data-name='Combined shape 7742'
        d='M7,18a.945.945,0,0,1-1-1V14H8v2h7V4H8V6H6V3A.945.945,0,0,1,7,2h9a.945.945,0,0,1,1,1V17a.945.945,0,0,1-1,1ZM1,10,5,6V9h6v2H5v3Z'
        transform='translate(-1 -2)'
      />
    </svg>
  </IconWrapper>
);

export default Logout;
