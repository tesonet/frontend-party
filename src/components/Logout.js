import React, { useEffect } from "react";
import { connect } from "react-redux";
import { logout } from "../store/actions/actions";
import { Redirect } from "react-router-dom";

const Logout = props => {
  useEffect(() => {
    props.onLogout();
  });

  return <Redirect to="/login" />;
};

const mapStateToProps = state => {
  return {};
};

const mapDispatchToProps = dispatch => {
  return {
    onLogout: () => dispatch(logout())
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Logout);
