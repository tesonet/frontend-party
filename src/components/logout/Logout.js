import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';

import style from './Logout.scss';
import { removeToken } from '../../api/auth-token';

class Logout extends Component {
  logout(e) {
    e.preventDefault();

    removeToken(); // todo: invalidate token on server?

    const { history } = this.props;
    history.push('/servers');
  }

  render() {
    return (
      <div className={style.logout}>
        <a href="/" onClick={(event) => { this.logout(event); }}>logout</a>
      </div>
    );
  }
}

Logout.defaultProps = {
  history: {},
};

Logout.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }),
};

export default withRouter(Logout);
