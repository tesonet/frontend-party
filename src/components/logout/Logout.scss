.logout {
  float: right;
  cursor: pointer;

  a {
    color: #2f3254;
    font-family: 'Roboto', sans-serif;
    font-weight: 300;
    font-size: 14px;
    line-height: 20px;
    padding: 10px;
    border: 1px solid transparent;

    &:hover {
      text-decoration: none;
      border: 1px solid #99cc33;
    }

    &:before {
      position: relative;
      top: 3px;
      content: '';
      background: transparent url(../../assets/images/icons/ico-logout.jpg) no-repeat;
      width: 16px;
      height: 16px;
      display: inline-block;
      margin-right: 12px;
    }
  }
}
