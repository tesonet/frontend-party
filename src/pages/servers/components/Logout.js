import React from 'react'

export const Logout = ({ handleLogout }) => (
    <div className="logout-button" onClick={handleLogout}>Logout</div>
)