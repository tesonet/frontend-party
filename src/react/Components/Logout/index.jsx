import {connect} from 'react-redux';
import LogoutButton from './logoutButton';

import {LOGOUT} from "../../Constants/Login";

const mapDispatchToProps = dispatch => {
    return {
        logoutRequest: (token) => {dispatch({type: LOGOUT, token: token})}
    }
};

export default connect(null, mapDispatchToProps)(LogoutButton);
