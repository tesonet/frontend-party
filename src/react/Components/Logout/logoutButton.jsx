import React from "react";
import PropTypes from "prop-types";
import {withRouter} from "react-router-dom";
import logoutIcon from "../../../express/public/static/img/logout/logout-ico.png"

class LogoutButton extends React.Component {
    constructor(props) {
        super(props);
        this.handleLogout = this.handleLogout.bind(this);
    }

    handleLogout(){
     localStorage.removeItem('auth_token');
     this.props.logoutRequest(null);
     this.props.history.push('/login');
    }

    render() {
        return (
            <div>
                <div className="btn-logout box" onClick={this.handleLogout}>
                    <img src={logoutIcon} className="logout-ico"/>
                    Logout
                </div>
            </div>
        )
    }
}

LogoutButton.propTypes = {
    logoutRequest: PropTypes.func
};

export default withRouter(LogoutButton);
