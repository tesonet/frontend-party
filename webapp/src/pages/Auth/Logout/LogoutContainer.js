// eslint-disable-next-line
import React, { Component } from 'react';
import { connect } from 'react-redux';

import privateRoute from '../../../hoc/privateRoute';
import api from '../../../utils/api';
import { doLogout } from '../authActions';

export const ROUTE_PATH = '/logout';

export class LogoutContainer extends Component {
  static defaultProps = {
    doLogout: () => {}
  };

  componentDidMount() {
    api.setToken(null);
    this.props.doLogout();
  }

  render() {
    return null;
  }
}

export default privateRoute(connect(null, { doLogout })(LogoutContainer));
