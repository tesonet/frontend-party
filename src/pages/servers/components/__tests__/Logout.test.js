import { Logout } from '../Logout';

it("should match snapshot", () => {
    const wrapper = shallow(
        <Logout handleLogout={() => {}}/>
    );
    expect(wrapper).toMatchSnapshot();
});