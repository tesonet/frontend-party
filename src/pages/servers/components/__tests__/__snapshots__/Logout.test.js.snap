// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="logout-button"
  onClick={[Function]}
>
  Logout
</div>
`;
